extensions/amp-autocomplete/0.1/amp-autocomplete.js

```diff
@@ -215,17 +227,38 @@ export class AmpAutocomplete extends AMP.BaseElement {
       );
     }
 
-    if (
-      this.templates_.hasTemplate(this.element, 'template, script[template]')
-    ) {
-      this.templateElement_ = this.templates_.findTemplate(
-        this.element,
-        'template, script[template]'
+    const isSsr = this.ssrTemplateHelper_.isSupported();
+    const hasTemplate = this.templates_.hasTemplate(
+      this.element,
+      'template, script[template]'
+    );
+    if (isSsr) {
+      userAssert(
+        hasTemplate,
+        `${TAG} should provide a <template> or <script type="plain/text"> element.`
       );
-      // Dummy render to verify existence of "data-value" attribute.
-      this.templates_
-        .renderTemplate(this.templateElement_, /** @type {!JsonObject} */ ({}))
+      userAssert(
+        !this.element.hasAttribute('filter'),
+        `${TAG} does not support client-side filter when server-side render is possible`
+      );
+      this.filter_ = FilterType.NONE;
+    } else {
+      this.filter_ = userAssert(
+        this.element.getAttribute('filter'),
+        `${TAG} requires "filter" attribute.`
+      );
+      userAssert(
+        isEnumValue(FilterType, this.filter_),
+        `Unexpected filter: ${this.filter_}`
+      );
+    }
+    if (hasTemplate) {
+      // Dummy render to verify existence of "data-value" or "data-disabled" attribute.
+      const empty = /** @type {!JsonObject} */ ({});
+      this.ssrTemplateHelper_
+        .applySsrOrCsrTemplate(this.element, empty)
```

Comment:
`data-value` tells us which property from the JsonObject to insert into the input field when the user selects a suggestion
`data-disabled` tells us not to interact with an item that might normally be an option (i.e. "bananas are sold out")

If neither of these are present and the user selects an item, the input field won't be able to reflect the choice.

But, now that I'm looking at this again, I'm not sure this block can verify that one of those attributes is always present, esp. since the dummy data is empty. We can move this check to later on when the rendered items get attached to the component, but I'm not sure what is the behavior if they are not present at that point.

Reply:
Yea, might be more useful as a `user().warn` in development mode.

Reply:
I'm not convinced this preliminary check in the `buildCallback` is useful anymore. I'm going to remove it since there is a similar `userAssert` at the time of render.